hbase-server/src/main/java/org/apache/hadoop/hbase/replication/ReplicationSourceOverallController.java

```diff
@@ -0,0 +1,31 @@
+package org.apache.hadoop.hbase.replication;
+
+import org.apache.hadoop.hbase.replication.regionserver.MetricsReplicationGlobalSourceSource;
+import org.apache.hadoop.hbase.replication.regionserver.RecoveredReplicationSource;
+import org.apache.yetus.audience.InterfaceAudience;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Used to control all replication sources inside one RegionServer or ReplicationServer.
+ * Used by {@link ReplicationSource} or {@link RecoveredReplicationSource}.
+ */
+@InterfaceAudience.Private
+public interface ReplicationSourceOverallController {
+
+  /**
+   * Returns the maximum size in bytes of edits held in memory which are pending replication
+   * across all sources inside this RegionServer or ReplicationServer.
+   */
+  long getTotalBufferLimit();
```

Comment:
Is this a 'size' rather than a 'limit'? It is count of how many bytes are currently accumulated in replication source memory?

Reply:
Maybe the comment is just in the wrong place.. should be on the next data member?

Reply:
This was copied from the ReplicationSourceManager. And it is the total limit, not the currently size.